File: app/src/main/java/fr/openium/examplepam/fragment/HistoryListFragment.java
package fr.openium.examplepam.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import fr.openium.examplepam.R;
import fr.openium.examplepam.adapter.HistoryAdapter;
import fr.openium.examplepam.database.AppDatabase;
import fr.openium.examplepam.model.Call;

public class HistoryListFragment extends Fragment {
    private RecyclerView recyclerViewCall;
    private TextView emptyTextView;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_call_history, container, false);
        recyclerViewCall = view.findViewById(R.id.recyclerViewContact);
        emptyTextView = view.findViewById(R.id.textViewEmpty);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        recyclerViewCall.setLayoutManager(new LinearLayoutManager(getContext()));
        displayCallSync();
    }

    private void displayCall() {
        List<Call> calls = AppDatabase.getInstance(getContext()).callDao().getAll();
        if (!calls.isEmpty()) {
            recyclerViewCall.setLayoutManager(new LinearLayoutManager(getContext()));
            recyclerViewCall.setAdapter(new HistoryAdapter(calls));
        } else {
            recyclerViewCall.setVisibility(View.GONE);
            emptyTextView.setVisibility(View.VISIBLE);
        }
    }

    private void displayCallSync() {
        AppDatabase.getInstance(getContext()).callDao().getAllSync().observe(this, new Observer<List<Call>>() {
            @Override
            public void onChanged(List<Call> calls) {
                if (!calls.isEmpty()) {
                    recyclerViewCall.setVisibility(View.VISIBLE);
                    emptyTextView.setVisibility(View.GONE);
                    if (recyclerViewCall.getAdapter() != null) {
                        ((HistoryAdapter) recyclerViewCall.getAdapter()).updateCalls(calls);
                    } else {
                        recyclerViewCall.setAdapter(new HistoryAdapter(calls));
                    }
                } else {
                    recyclerViewCall.setVisibility(View.GONE);
                    emptyTextView.setVisibility(View.VISIBLE);
                }
            }
        });
    }

//    private void displayContacts() {
//        linearLayout.setVisibility(View.GONE);
//        recyclerViewContact.setLayoutManager(new LinearLayoutManager(getContext()));
//        List<CallContact> contacts = getContactName();
//        recyclerViewContact.setAdapter(new ContactAdapter(contacts, (contact -> {
//
//        })));
//    }
//
//    private List<String> getContactName() {
//        ContentResolver cr = getContext().getContentResolver();
//        Cursor cur = cr.query(ContactsContract.Contacts.CONTENT_URI, null, null, null, null);
//        List<String> contacts = new ArrayList<>();
//
//        if ((cur != null ? cur.getCount() : 0) > 0) {
//            while (cur.moveToNext()) {
//                String name = cur.getString(cur.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
//                contacts.add(name);
//            }
//        }
//        if (cur != null) {
//            cur.close();
//        }
//
//        return contacts;
//    }

}

File: app/src/main/java/fr/openium/examplepam/ui/history/HistoryListFragment.kt
package fr.openium.examplepam.ui.history

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import fr.openium.examplepam.R
import fr.openium.examplepam.adapter.HistoryAdapter
import fr.openium.examplepam.model.Call

class HistoryListFragment : Fragment() {
    private var recyclerViewCall: RecyclerView? = null
    private var emptyTextView: TextView? = null
    private val viewModel: HistoryViewModel by viewModels()
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_call_history, container, false)
        recyclerViewCall = view.findViewById(R.id.recyclerViewContact)
        emptyTextView = view.findViewById(R.id.textViewEmpty)
        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        recyclerViewCall?.layoutManager = LinearLayoutManager(context)
        viewModel.calls.observe(viewLifecycleOwner) {
            displayCalls(it)
        }
    }

    private fun displayCalls(calls: List<Call>) {
        if (!calls.isEmpty()) {
            emptyTextView?.visibility = View.GONE

            recyclerViewCall?.apply {
                //Here this is the recyclerViewCall
                visibility = View.VISIBLE
                if (adapter != null) {
                    (adapter as HistoryAdapter?)?.updateCalls(calls)
                } else {
                    adapter = HistoryAdapter(calls)
                }
            }

        } else {
            emptyTextView?.visibility = View.VISIBLE
            recyclerViewCall?.visibility = View.GONE
        }
    }
}